services/ux-backend/handlers/onboardingtokens/handler.go

```diff
@@ -13,17 +14,49 @@ const (
 	onboardingPrivateKeyFilePath = "/etc/private-key/key"
 )
 
+var unitToGib = map[string]int{
+	"Gi": 1,
+	"Ti": 1024,
+	"Pi": 1024 * 1024,
+}
+
 func HandleMessage(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
 	switch r.Method {
 	case "POST":
-		handlePost(w, tokenLifetimeInHours)
+		handlePost(w, r, tokenLifetimeInHours)
 	default:
 		handleUnsupportedMethod(w, r)
 	}
 }
 
-func handlePost(w http.ResponseWriter, tokenLifetimeInHours int) {
-	if onboardingToken, err := util.GenerateOnboardingToken(tokenLifetimeInHours, onboardingPrivateKeyFilePath); err != nil {
+func handlePost(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
+	var storageQuotaInGiB *int
+	// When ContentLength is 0 that means request body is empty and
+	// storage quota is unlimited
+	var err error
+	if r.ContentLength != 0 {
+		var quota = struct {
+			Value int    `json:"value"`
+			Unit  string `json:"unit"`
+		}{}
+		if err = json.NewDecoder(r.Body).Decode(&quota); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		if quota.Value <= 0 {
+			http.Error(w, fmt.Sprintf("invalid Value sent in request body: %v", quota.Value), http.StatusBadRequest)
+			return
+		}
+
+		unitAsGiB, ok := unitToGib[quota.Unit]
+		if !ok {
+			http.Error(w, fmt.Sprintf("invalid Unit type sent in request body, Valid types are [Gi,Ti,Pi]: %v", quota.Unit), http.StatusBadRequest)
+			return
+		}
+		val := unitAsGiB * quota.Value
+		storageQuotaInGiB = &val
```

Comment:
No need for intermiddate var
```suggestion
		storageQuotaInGiB = ptr.To(unitAsGiB * quota.Value)
```